code/obj/item/clothing/helmets.dm

```diff
@@ -687,15 +690,28 @@
 		see_face = TRUE
 		icon_state = "welding-up"
 		boutput(user, "You flip the mask up. The mask is now providing greater armor to your head.")
-		color_r = 1 // no effect
-		color_g = 1
-		color_b = 1
+		src.reveal(user)
 		user.update_clothing()
 
 		src.c_flags &= ~(COVERSEYES | BLOCKCHOKE)
 		setProperty("meleeprot_head", 4)
 		setProperty("disorient_resist_eye", 0)
 
+	equipped(mob/user, slot)
+		. = ..()
+		if (!src.up)
+			src.obscure(user)
+
+	unequipped(mob/user)
+		. = ..()
+		src.reveal(user)
+
+	disposing()
+		. = ..()
+		if (ismob(src.loc))
+			var/mob/owner = src.loc
+			src.reveal(owner)
```

Comment:
hm. If you're wearing a weldingmask, and also have one in your hand, would pouring acid on the one in your hand clear the overlay b/c of this? Might be worth testing

Reply:
This. Likely checks for if it is in slot.